container-search/src/main/java/com/yahoo/search/handler/SearchHandler.java

```diff
@@ -290,14 +291,14 @@ private HttpSearchResponse handleBody(HttpRequest request){
         // Create query
         Query query;
         if (request.getMethod() == com.yahoo.jdisc.http.HttpRequest.Method.POST && request.getHeader(com.yahoo.jdisc.http.HttpHeaders.Names.CONTENT_TYPE).equals(JSON_CONTENT_TYPE)) {
-            Inspector inspector = null;
+            Inspector inspector;
             try {
                 byte[] byteArray = IOUtils.readBytes(request.getData(), 1 << 20);
                 inspector = SlimeUtils.jsonToSlime(byteArray).get();
```

Comment:
This particular line is what throws on bad JSON, so it needs to be in a separate try-catch if you are to separate the two problems.